Samples~/Encryption/EncryptionSample.cs

```diff
@@ -0,0 +1,571 @@
+                encryptKeyArray.Dispose();
+            if(bytes.Length > 0)
+                encryptKeyArray = new NativeArray<byte>(bytes, Allocator.Persistent);
+        }
+    }
+
+    private void OnChangedDecryptKeyInput(string data)
+    {
+        byte[] bytes = System.Text.Encoding.UTF8.GetBytes(data);
+
+        lock (decryptKeyInputLock)
+        {
+            if (decryptKeyArray.IsCreated)
+                decryptKeyArray.Dispose();
+            if (bytes.Length > 0)
+                decryptKeyArray = new NativeArray<byte>(bytes, Allocator.Persistent);
+        }
+    }
+
+    private void OnDestroy()
+    {
+        if (encryptKeyArray.IsCreated)
+            encryptKeyArray.Dispose();
+        if (decryptKeyArray.IsCreated)
+            decryptKeyArray.Dispose();
+        if (encryptedDataArray.IsCreated)
+            encryptedDataArray.Dispose();
+        if (decryptedDataArray.IsCreated)
+            decryptedDataArray.Dispose();
+    }
+
+    private void SetUpSenderTransform(RTCRtpSender sender)
+    {
+        sender.Transform = new RTCRtpScriptTransform(TrackKind.Video, e => OnSenderTransform(sender.Transform, e));
+    }
+
+    private void SetUpReceiverTransform(RTCRtpReceiver receiver)
+    {
+        receiver.Transform = new RTCRtpScriptTransform(TrackKind.Video, e => OnReceiverTransform(receiver.Transform, e));
+    }
+
+    void OnSenderTransform(RTCRtpTransform transform, RTCTransformEvent e)
+    {
+        lock (encryptKeyInputLock)
+        {
+            if(encryptKeyArray.IsCreated)
+            {
+                if(e.Frame is not RTCEncodedVideoFrame frame)
+                {
+                    return;
+                }
+
+                var data = frame.GetData();
+
+                // resize NativeArray
+                if (data.Length > encryptedDataArray.Length)
+                {
+                    if (encryptedDataArray.IsCreated)
+                        encryptedDataArray.Dispose();
+                    encryptedDataArray = new NativeArray<byte>(data.Length, Allocator.Persistent, NativeArrayOptions.UninitializedMemory);
+                }
+
+                var cryptoOffset = frameTypeToCryptoOffset[frame.Type];
+                NativeArray<byte>.Copy(src: data, srcIndex: 0, dst: encryptedDataArray, dstIndex: 0, length: cryptoOffset);
+
+                for(int i = cryptoOffset; i < data.Length; i++)
+                {
+                    var key = encryptKeyArray[i % encryptKeyArray.Length];
+                    encryptedDataArray[i] = (byte)(data[i] ^ key);
+                }
+                e.Frame.SetData(encryptedDataArray.AsReadOnly(), 0, data.Length);
+            }
+        }
+        transform.Write(e.Frame);
+    }
+
+    void OnReceiverTransform(RTCRtpTransform transform, RTCTransformEvent e)
+    {
+
+        lock (decryptKeyInputLock)
+        {
+            if (decryptKeyArray.IsCreated)
+            {
+                if (e.Frame is not RTCEncodedVideoFrame frame)
```

Comment:
```suggestion
                if (!(e.Frame is RTCEncodedVideoFrame frame))
```
same as https://github.com/Unity-Technologies/com.unity.webrtc/pull/898/files#diff-4b5680394927d27eb23ee24f50df9a36851d684113eb0e1c120dc0075170c46bR188